projects/kit/components/progress/progress-circle/progress-circle.style.less

```diff
@@ -1,96 +1,142 @@
 @import '@taiga-ui/core/styles/taiga-ui-local';
 
 /* Safari doesn't support rem units inside svg (but supports em units) */
-@width: {
+@diameter: {
+    xxs: 2rem;
+    xs: 2.5rem;
+    s: 3.5rem;
+    m: 4rem;
+    l: 5rem;
+    xl: 6rem;
+    xxl: 8rem;
+};
+
+// TODO: delete in v4.0
+@legacy-width: {
     @s: 2em;
     @m: 3em;
     @l: 4em;
     @xl: 7em;
 };
 
-@track-stroke: {
+// TODO: delete in v4.0
+@legacy-track-stroke: {
     @s: 0.5em;
     @m: 0.375em;
     @l: 0.375em;
     @xl: 0.375em;
 };
 
-@progress-stroke: {
+// TODO: delete in v4.0
+@legacy-progress-stroke: {
     @s: 0.5em;
     @m: 0.375em;
     @l: 0.375em;
     @xl: 0.375em;
 };
 
-.circle-params(@size) {
-    width: @width[@@size];
-    height: @width[@@size];
-
-    .t-track {
-        /* clean-css ignore:start */
-        r: ((@width[@@size] - @track-stroke[@@size]) / 2);
-
-        /* clean-css ignore:end */
-        stroke-width: @track-stroke[@@size];
-    }
-
-    .t-progress {
-        @radius: ((@width[@@size] - @progress-stroke[@@size]) / 2);
-        @circumference: 2 * pi() * @radius;
-
-        /* clean-css ignore:start */
-        r: @radius;
-
-        /* clean-css ignore:end */
-        stroke-width: @progress-stroke[@@size];
-        stroke-dasharray: @circumference;
-        stroke-dashoffset: @circumference;
-
-        &_filled {
-            transition: stroke-dashoffset var(--tui-duration) linear;
-            stroke-dashoffset: calc(@circumference - var(--progress-ratio) * @circumference);
-        }
-    }
-}
-
 :host {
     position: relative;
     display: block;
     color: var(--tui-primary);
     transform: rotate(-90deg);
     transform-origin: center;
-    font-size: 1rem;
+    font-size: 1rem; // to use em units inside svg (Safari doesn't support rem units inside svg)
+    width: var(--t-diameter);
+    height: var(--t-diameter);
 
     &[data-mode='onDark'] {
         --tui-clear: var(--tui-clear-inverse);
     }
 
     &[data-size='s'] {
-        .circle-params(s);
+        --t-diameter: @legacy-width[@s];
+        --t-track-stroke: @legacy-track-stroke[@s];
+        --t-progress-stroke: @legacy-progress-stroke[@s];
     }
 
     &[data-size='m'] {
-        .circle-params(m);
+        --t-diameter: @legacy-width[@m];
+        --t-track-stroke: @legacy-track-stroke[@m];
+        --t-progress-stroke: @legacy-progress-stroke[@m];
     }
 
     &[data-size='l'] {
-        .circle-params(l);
+        --t-diameter: @legacy-width[@l];
+        --t-track-stroke: @legacy-track-stroke[@l];
+        --t-progress-stroke: @legacy-progress-stroke[@l];
     }
 
     &[data-size='xl'] {
-        .circle-params(xl);
+        --t-diameter: @legacy-width[@xl];
+        --t-track-stroke: @legacy-track-stroke[@xl];
+        --t-progress-stroke: @legacy-progress-stroke[@xl];
+    }
+
+    &[new] {
+        --t-track-stroke: var(--tui-thickness, 0.375em);
+        --t-progress-stroke: var(--tui-thickness, 0.375em);
```

Comment:
Guys, should customization of thickness should be just css-variable or `@Input() thickness = 0.375` ? 🤔 

![Screenshot 2023-11-14 at 17 16 31](https://github.com/taiga-family/taiga-ui/assets/35179038/2992d736-b0b4-4ffd-ba98-e03916d04b3c)

Reply:
I think it's best to keep it a CSS variable if it only affects styles.